src/connector/src/sink/redis.rs

```diff
@@ -38,35 +90,296 @@ impl Sink for RedisSink {
     type Writer = RedisSinkWriter;
 
     async fn new_writer(&self, _writer_env: SinkWriterParam) -> Result<Self::Writer> {
-        todo!()
+        Ok(RedisSinkWriter::new(
+            self.config.clone(),
+            self.schema.clone(),
+            self.pk_indices.clone(),
+            self.is_append_only,
+        )?)
     }
 
     async fn validate(&self, _client: Option<ConnectorClient>) -> Result<()> {
```

Comment:
I think we should validate the key_foramt and value_format here to make sure the user-provided format matches with our spec. For example, the field name matches with an existing column in the schema; the field name is wrapped by `{}`, etc.